### ISSUE TITLE
Clarification: Differential from datadog_flutter

### ISSUE THREAD

**Issue by mikhael28** [DataDog/dd-sdk-flutter#174]
Hi there Datadog! great to see there is now an 'official' library - as a user of the datadog community implementation package, could someone please describe the practical differences between this package and that one? Is it worth switching right now? 

Additionally, are there are any regressions at the moment between that library and this one? And, what is on the roadmap for this particular package? I haven't seen any package updates in... 34 days now.

**Comment by scopendo** [comment]
Suggest reading https://github.com/GetDutchie/datadog_flutter/issues/108

**Comment by fuzzybinary** [comment]
Hi @mikhael28,

So far as we know, there are no regressions between the community library and this one, other than the removal of Span support, and we do not officially support Flutter Web (although I believe what we support currently is in line with what the Community SDK supported)

A lot of the support from the community SDK works similarly in this one, including automatic view tracking using the DatadogNavigationObserver.

Some practical differences right now:
- Up to date on native dependencies and their bug fixes (see the CHANGELOG for more details)
- Better property serialization support.
- The tracking http client (`datadog_tracking_http_client` package) tracks at a lower level than the community plugin, which means it will track resource requests from Flutter itself (like the NetworkImage widget) as well as from packages like Dio without any additional configuration
- The Datadog UI will show a Flutter icon, as well as the platform. You can use the same Application Id for both platforms and still differentiate between iOS and Android sessions.
- Improved native crash tracking for both iOS and Android
- Messages from the native Datadog SDKs will be shown in your Flutter debug console if you set sdkVerbosity.
- Official support from Datadog!

As for updates -- after the initial flurry of releases, we're gathering feedback for future improvement and making sure we have a reasonable amount of fixes and new features before releasing a new version. Take that 34 days to mean we've been stable during that time and haven't needed to rush out any fixes.

As for roadmap -- Datadog is very interested to hear from people using the library about what features you think are most important. If there's something you want on our roadmap, please reach out to support or your sales rep and let them know what you're most interested in.

**Comment by mikhael28** [comment]
Alright, that sounds good - I have it in my backlog to migrate to the official library in the next 3 weeks. Will report back with any feedback.

**Comment by mikhael28** [comment]
Hi there! A few quick things about the experience migrating to Datadog's official Flutter SDK.

1. It was overall pretty smooth, but I was using Datadog in an incredibly limited way - I was just logging errors, with no RUM events or proactive logging. This segues into a big problem I found, which may the unofficial SDK may also be suffering from but I just don't know.
2. When you are trying to manually send a DD log, if you try to use the DD object outside of the closure for the top level widget context object, it will not work (but it also will not throw an error). For example, let's say you have a `api_service.dart` class which just has some methods for triggering an http request. It is outside of a Widget, and as a result the DD log send silently fails.
3. Or, for example, if I have a nested Widget, inside a nested widget, inside the top level widget. Three Widgets deep. If I try to use the DD logger in the top level widget, it works. If I try to use the logger one layer deep, it works. If I try to use it two layers deep, it silently fails. In my opinion, this is a problem as I have some relatively complex conditional rendering.

Does this sound familiar? Is this a known issue?

**Comment by fuzzybinary** [comment]
Hi @mikhael28,

No that does not sound familiar and would definitely be a bug if that is the case.

Can you turn on verbose SDK logging and see if maybe the error isn't so silent? Before Datadog initialization call:
```dart
DatadogSdk.sdkVerbosity = Verbosity.verbose;
```

If that doesn't give you any info, if you could send me over some sample code that reproduces the issue I'll take a look.

Thanks!

**Comment by mikhael28** [comment]
Hey @fuzzybinary - hope all is well. I installed the latest version of the SDK today, and I am now able to fire off logs from a generic class. I have not tested my triple-nested widget yet, but I don't need to anymore necessarily. The web version was not working in the 1.0.2 release for me, but I added it back in 1.0.3 and it seems to be good for me now!

All in all, I consider myself 'migrated' from the unofficial SDK. Thank you for your help and support!
